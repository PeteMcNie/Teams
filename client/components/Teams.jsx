import React from 'react'
import { connect } from 'react-redux'

export const Teams = () => {
  return (
    <div>
      <h3>hi</h3>
    </div>
  )
}

const mapStateToProps = state => {
  console.log('Teams: ', state)
  return {
    teams: state
  }
}

export default connect(mapStateToProps)(Teams)
