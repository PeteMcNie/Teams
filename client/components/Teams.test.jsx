import React from 'react'

import { renderWithRedux } from '../testing/utils'
import { screen } from '@testing-library/react'

import '@testing-library/jest-dom'

import Teams from './Teams'

const newTeams = [
  [
    { id: 1, name: 'Pete', isSelected: 0 },
    { id: 55, name: 'Steve', isSelected: 0 },
    { id: 4, name: 'Jill', isSelected: 0 }
  ],
  [
    { id: 23, name: 'Dan', isSelected: 0 },
    { id: 6, name: 'Lucy', isSelected: 0 },
    { id: 87, name: 'Court', isSelected: 0 }
  ],
  [
    { id: 3, name: 'Chloe', isSelected: 0 },
    { id: 86, name: 'Nigel', isSelected: 0 },
    { id: 21, name: 'Nico', isSelected: 0 }
  ]
]

test('teams component matches snapshot', () => {
  const { asFragment } = renderWithRedux(<Teams />,
    { initialState: { newTeams } })

  expect(asFragment()).toMatchSnapshot()
})

test('teams component renders the correct number of teams', () => {
  renderWithRedux(<Teams />, { initialState: { newTeams } })

  expect(screen.getByText('Nigel')).toBeInTheDocument()
  expect(screen.getByText('Court')).toBeInTheDocument()
  expect(screen.getByText('Pete')).toBeInTheDocument()
})

test('teams component renders the reshuffle link correctly', async () => {
  renderWithRedux(<Teams />, { initialState: { newTeams } })

  const link = await screen.getByTestId('link')
  expect(link).toBeInTheDocument()
  expect(screen.getByText('Re-shuffle')).toBeInTheDocument()
})
