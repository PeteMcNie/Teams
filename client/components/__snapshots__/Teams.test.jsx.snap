// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`teams component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="team"
    >
      <h3
        class="teamHeading"
      >
        Team One:
      </h3>
      <ul
        class="teamsGrid"
      >
        <li>
          Pete
        </li>
        <li>
          Steve
        </li>
        <li>
          Jill
        </li>
      </ul>
    </div>
    <div
      class="team"
    >
      <h3
        class="teamHeading"
      >
        Team Two:
      </h3>
      <ul
        class="teamsGrid"
      >
        <li>
          Dan
        </li>
        <li>
          Lucy
        </li>
        <li>
          Court
        </li>
      </ul>
    </div>
    <div
      class="team"
    >
      <h3
        class="teamHeading"
      >
        Team Three:
      </h3>
      <ul
        class="teamsGrid"
      >
        <li>
          Chloe
        </li>
        <li>
          Nigel
        </li>
        <li>
          Nico
        </li>
      </ul>
    </div>
    <div
      class="reshufflediv"
      data-testid="link"
    >
      <a
        class="reshuffle"
        href="/createTeams"
      >
        Re-shuffle
      </a>
    </div>
  </div>
</DocumentFragment>
`;
